File: web/src/app/Models/Dtos/Question/create-question-request-dto.ts
import { DifficultyLevel } from '../../Enums/DifficultyLevel';
import { QuestionType } from '../../Enums/QuestionType';
import { CodeCompletionQuestionPayloadDto, FillInTheBlankQuestionPayloadDto, MultipleChoiceQuestionPayloadDto, OpenEndedQuestionPayloadDto, TrueFalseQuestionPayloadDto} from './question-type-payload-dtos';

export class CreateQuestionRequestDto {
  type: QuestionType = QuestionType.MultipleChoice;
  language?: string;
  difficulty: DifficultyLevel = DifficultyLevel.Junior;
  tags: string[] = [];
  title = '';
  questionText = '';
  createdBy = '';

  multipleChoice?: MultipleChoiceQuestionPayloadDto;
  codeCompletion?: CodeCompletionQuestionPayloadDto;
  openEnded?: OpenEndedQuestionPayloadDto;
  // TODO(OpenEnded-cleanup): remove legacy wire property after backend contract rename.
  fillInTheBlank?: FillInTheBlankQuestionPayloadDto;
  trueFalse?: TrueFalseQuestionPayloadDto;
}

File: backend/SkillProof/SkillProof.Entities/Dtos/Questions/CreateQuestionRequestDto.cs
using SkillProof.Entities.Enums;
using System.ComponentModel.DataAnnotations;

namespace SkillProof.Entities.Dtos.Questions
{
    public class CreateQuestionRequestDto
    {
        [Required]
        public QuestionType Type { get; set; }

        [Required]
        [StringLength(20)]
        public string Language { get; set; } = string.Empty;

        [Required]
        public DifficultyLevel Difficulty { get; set; }

        public List<string> Tags { get; set; }

        [Required]
        [StringLength(255)]
        public string Title { get; set; } = string.Empty;

        [Required]
        public string QuestionText { get; set; } = string.Empty;

        [Required]
        [StringLength(450)]
        public string CreatedBy { get; set; } = string.Empty;

        public MultipleChoiceQuestionPayloadDto? MultipleChoice { get; set; }
        public CodeCompletionQuestionPayloadDto? CodeCompletion { get; set; }
        public FillInTheBlankQuestionPayloadDto? FillInTheBlank { get; set; }
        public TrueFalseQuestionPayloadDto? TrueFalse { get; set; }
    }
}
